src/runtime/providers/string.ts

```diff
@@ -251,7 +251,83 @@ class Urlencode extends Constraint {
   }
 }
 
+class Length extends Constraint {
+  static AttributeMapping = {
+    "text": 0,
+    "as": 1,
+  }
+
+  validAsOption(az) {
+    if (az === undefined || az === "symbols" || az === "code-points") {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  getLength(text, az) {
+    if (az === "symbols") {
+      return [this.countSymbols(text)];
+    } else if (az === "code-points") {
+      return [text.length];
+    }
+    return undefined;
+  }
+
+  resolveProposal(proposal, prefix) {
+    let {args} = this.resolve(prefix);
+    let [text, az] = args;
+    if (az === undefined) {az = "symbols"};
```

Comment:
Let's either drop the braces or indent this appropriately. This looks confusingly like an object.